pkg/controller/whisker/whisker_controller.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package whisker
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	"github.com/tigera/operator/pkg/ctrlruntime"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/monitor"
+	"github.com/tigera/operator/pkg/render/whisker"
+)
+
+const (
+	controllerName = "whisker-controller"
+	ResourceName   = "whisker"
+)
+
+var log = logf.Log.WithName(controllerName)
+
+// Add creates a new Reconciler Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and start it when the Manager is started.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.WhiskerEnabled {
```

Comment:
This will probably prevent us from uninstalling Whisker in EE, since presumably the CRD won't exist there?

Reply:
Yes for right now, I wanted to follow up with uninstalling in a follow up PR. Right now nobody can install this, so I think follow up should be fine (if you agree).

Reply:
Forget what I said, I did the thing.